xwiki-platform-core/xwiki-platform-sharepage/xwiki-platform-sharepage-ui/src/main/resources/XWiki/SharePage.xml

```diff
@@ -337,7 +337,7 @@ $!{escapetool.xml($message)}
 &lt;/pre&gt;
 #end
 
-&lt;p&gt;To visit this document online, open &lt;a href="${doc.getExternalURL()}"&gt;${doc.getExternalURL()}&lt;/a&gt; in your browser.&lt;/p&gt;
+&lt;p&gt;To visit this document online, open &lt;a href="${doc.getExternalURL()}"&gt;${util.decodeURI("${doc.getExternalURL()}")}&lt;/a&gt; in your browser.&lt;/p&gt;
```

Comment:
Define "encoded"... The whole URL isn't encoded in any way, but parts of it are and should be `%XY` encoded, according to the URL grammar, so that illegal characters aren't used. For example, spaces in the page name, unencoded, will break the URL, so will commas, `<` and `>`, and so on. `doc.get*URL` does proper escaping of the fragments, and there shouldn't be any other escaping as far as I can see. So, I really don't see what kind of decoding is needed.